### PR TITLE
Use a stable RNG in fakedata

### PR DESCRIPTION
**Pull request by t-bltg**
Failures have been seen on nightly, change to a more stable random number generator.

@BeastyBlacksmith, if this is OK for you I will re-generate ref images.

### CODE REVIEW

**Comment by BeastyBlacksmith** [comment]
You'd need to move StableRNG from `[extras]` to `[deps]` in `Porject.toml`, no?

**Comment by t-bltg** [comment]
> You'd need to move StableRNG from `[extras]` to `[deps]` in `Project.toml`, no?

Yes, done.

**Comment by BeastyBlacksmith** [comment]
Thogh, I'd prefer that this stays a testdependency and you only allow passing a rng to `fakedate` and then pass the stable rng in the tests

**Comment by t-bltg** [comment]
> Thogh, I'd prefer that this stays a testdependency and you only allow passing a rng to `fakedate` and then pass the stable rng in the tests

IMO, `Plots.fakedata()` is exposed to the user throughout `src/examples.jl`, so moving `StableRNGs` from `extras` to `deps` is justified.

**Comment by t-bltg** [comment]
@BeastyBlacksmith all the test run with https://github.com/JuliaPlots/PlotReferenceImages.jl/pull/106.
~~I've also replaced `download` (deprecated) with `Downloads.download`.~~

File: src/utils.jl [hunk 1]
```diff
@@ -303,10 +303,10 @@ function convert_to_polar(theta, r, r_extrema = ignorenan_extrema(r))
     x, y
 end
 
-function fakedata(sz...)
+function fakedata(sz...; rng=StableRNG(PLOTS_SEED))
```
Comment (BeastyBlacksmith) [inline src/utils.jl:306]:
```suggestion
function fakedata(sz...; rng=Random.GLOBAL_RNG(PLOTS_SEED))
```
What I wanted to say is that we can keep using the unstable RNG here, and switch to the StableRNG in the tests

**Comment by t-bltg** [comment]
Should we then hack again examples in `replace_rand!` to insert a keyword argument or should we modify the examples using `Plots.fakedata` ?

**Comment by BeastyBlacksmith** [comment]
I'd shadow it in the tests like `fakedata(args...) = Plots.fakedata(args...; rng=StableRNG(Plots.PLOTS_SEED))`

**Comment by t-bltg** [comment]
> I'd shadow it in the tests like `fakedata(args...) = Plots.fakedata(args...; rng=StableRNG(Plots.PLOTS_SEED))`

Yeah that's what I came up with.

File: Project.toml [hunk 2]
```diff
@@ -31,6 +31,7 @@ SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
 UUIDs = "cf7118a7-6976-5b1a-9a39-7adc72f591a4"
+StableRNGs = "860ef19b-820b-49d6-a774-d7a799459cd3"
```
Comment (BeastyBlacksmith) [inline Project.toml:34]: That can go to [extras] then

File: src/Plots.jl [hunk 3]
```diff
@@ -11,6 +11,7 @@ using Reexport
 import GeometryBasics
 using Dates, Printf, Statistics, Base64, LinearAlgebra, Random
 using SparseArrays
+using StableRNGs
```
Comment (BeastyBlacksmith) [inline src/Plots.jl:14]:
```suggestion
```

File: test/imgcomp.jl [hunk 4]
```diff
@@ -41,6 +41,7 @@ function image_comparison_tests(
         expr = Expr(:block)
         append!(expr.args, example.exprs)
         fix_rand!(expr)
+        @debug expr
```
Comment (BeastyBlacksmith) [inline test/imgcomp.jl:44]: What is that for?

Comment (t-bltg) [reply test/imgcomp.jl:44]: Debugging in CI, if needed.

File: Project.toml (hunk 2 above)
Comment (t-bltg) [reply Project.toml]: Yes

File: src/Plots.jl (hunk 3 above)
Comment (t-bltg) [reply src/Plots.jl]: Done

**Review by BeastyBlacksmith — APPROVED** [review]
Still needs new reference figures

**Comment by t-bltg** [comment]
> Still needs new reference figures

Yes, there are here: https://github.com/JuliaPlots/PlotReferenceImages.jl/pull/106. I will merge them now, if you're ok with that.

**Comment by t-bltg** [comment]
Failure are related to `GraphRecipes`, made a PR there: https://github.com/JuliaPlots/GraphRecipes.jl/pull/162.

**Pull request merged**